src/startServerAndCreateNextHandler.ts

```diff
@@ -59,24 +66,27 @@ function startServerAndCreateNextHandler<
       return;
     }
 
-    const body = [];
+    let responseObject: string | ReadableStream;
 
     if (httpGraphQLResponse.body.kind === 'complete') {
-      body.push(httpGraphQLResponse.body.string);
+      responseObject = httpGraphQLResponse.body.string;
     } else {
-      for await (const chunk of httpGraphQLResponse.body.asyncIterator) {
-        body.push(chunk);
-      }
-    }
-
-    const headers: Record<string, string> = {};
-
-    for (const [key, value] of httpGraphQLResponse.headers) {
-      headers[key] = value;
+      const responseBody = httpGraphQLResponse.body;
+      responseObject = new ReadableStream({
+        async start(controller) {
+          for await (const chunk of responseBody.asyncIterator) {
+            controller.enqueue(chunk);
+          }
+          controller.close();
+        },
```

Comment:
I think it's more common to use pull, and then use the iterator without looping like this:
```suggestion
        async pull(controller) {
          if (httpGraphQLResponse.body.kind === 'chunked') {
            const { value, done } = await httpGraphQLResponse.body.asyncIterator.next();
        
            if (done) {
              controller.close();
            } else {
              controller.enqueue(value);
            }
          }
        },
```